AE-1108 Make read-only APIs visible for Laskuttaja

muep opened the pull request:
This change makes some of the ETP API available to users who have
logged in as a laskuttaja.

muep committed:
AE-1108 Make read-only APIs visible for Laskuttaja

This change makes some of the ETP API available to users who have
logged in as a laskuttaja.

muep commented:
Tämä ei nyt varmaan ole vielä ihan valmis eikä minusta välttämättä kannata mergetä ennen kuin fronttipuolelle on myös tehtynä jotain mikä tästä hyötyy.

muep commented on etp-backend/src/main/clj/solita/etp/api/energiatodistus.clj:
```diff
@@ -32,6 +32,8 @@
                         {:type :invalid-arguments :response 400}
                         {:type :schema.core/error :response 400}])
 
+(def et-get-access (some-fn rooli-service/laatija? rooli-service/paakayttaja? rooli-service/laskuttaja?))
```
Tämä on nyt myös siellä crud-moduulissa. En vielä yhdistellyt kun en ollut varma, mihin se sitten kuuluisi ja halutaanko ylipäänsä näitä toistuvia accesseja koota vakioksi. Jos on toiveita että miten näille tehdään, niin mieluusti kuulen niistä.

hkorpi replied:
ehkä tämän voisi laittaa rooli-serviceen esim. nimellä energiatodistus-reader? vähän samaan tyyliin kuin laatija-reader?

muep replied:
nyt pitäisi olla tämän ehdotuksen mukainen

hkorpi commented on etp-backend/src/main/clj/solita/etp/service/rooli.clj:
```diff
@@ -24,4 +24,8 @@
 
+(defn laatija-reader? [{:keys [rooli] :as whoami}]
+  (or (laatija-maintainer? whoami)
+      (contains? #{3} rooli)))
```
(laskuttaja? whoami) ;kolmen sijasta mielellään käytetään aina tuota predikaattia

hkorpi replied:
ja tämä koko funktion voi kirjoittaa suoraan (def laatija-reader? (some-fn laatija-maintainer? laskuttaja?))

hkorpi replied:
samoin laatija-maintainer? (some-fn paakayttaja? p...t...?)

muep replied:
tämä taitaa jäädä kokonaan pois nyt kun yhdistelee uusimpaan `develop`iin

muep committed:
AE-1108 Viesti functionality for Laskuttaja

Laskuttaja users are given their own group similarly to how laatija
users are handled.

MattiNieminen committed:
AE-1108 fix laskuttaja role related problems caused by rebase to develop

hkorpi commented on etp-backend/src/main/sql/solita/etp/db/viesti.sql:
```diff
@@ -25,6 +25,22 @@ where from_id = :laatija-id or vastaanottajaryhma_id = 1 or
 order by viestiketju.id desc
 limit :limit offset :offset;
 
+-- name: select-viestiketjut-for-laskuttaja
```
helpompi muokata: select-viestiketjut-for-laatija -> select-viestiketjut-for-kayttaja ja lisätä bind-parametri ryhmälle

hkorpi replied:
eli bind-parametrit olisivat: :kayttaja-id + :vastaanottajaryhma-id

hkorpi replied:
sitten funktio: rooli-id->vastaanottajaryhma-id

MattiNieminen replied:
Hyvä idea. Pistin tähän päälle commitin joka yhdistää nuo funkkarit.

MattiNieminen committed:
AE-1108 merge two similar db functions into one

muep commented:
Minun puolesta nuo Matin lisäämät muutokset on ok.

MattiNieminen committed:
AE-1108 implement possibility for laskuttaja to send viesti to single vastaanottaja
